File: src/main/java/me/centralhardware/znatoki/telegram/statistic/validate/AmountValidator.java
package me.centralhardware.znatoki.telegram.statistic.validate;

import com.google.common.primitives.Ints;
import io.vavr.control.Either;
import org.springframework.stereotype.Component;

@Component
public class AmountValidator implements Validator<String, Integer> {

    private static final String NOT_NUMBER_ERROR = "Введенное значение должно быть числом";
    private static final String LESS_THAN_ZERO_ERROR = "Введенное значение должно быть больше нуля";

    @Override
    public Either<String, Integer> validate(String value) {
        Integer parsedValue = Ints.tryParse(value);
        if (parsedValue == null) {
            return Either.left(NOT_NUMBER_ERROR);
        }
        if (parsedValue <= 0) {
            return Either.left(LESS_THAN_ZERO_ERROR);
        }
        return Either.right(parsedValue);
    }
}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/validate/AmountValidator.kt
package me.centralhardware.znatoki.telegram.statistic.validate

import arrow.core.Either
import org.springframework.stereotype.Component

@Component
class AmountValidator : Validator<String, Int> {

    override fun validate(value: String): Either<String, Int> {
        val parsedValue = value.toIntOrNull()
        return when {
            parsedValue == null -> {
                Either.Left("Введенное значение должно быть числом")
            }

            parsedValue <= 0 -> {
                Either.Left("Введенное значение должно быть больше нуля")
            }

            else -> {
                Either.Right(parsedValue)
            }
        }
    }
}